src/mfa.js

```diff
@@ -0,0 +1,96 @@
+import axios from "axios";
+import { setCookiesAndTokens } from "./cookies.js";
+import { store } from "./store.js";
+import { getQueryAttr, redirectToPath } from "./url.js";
+import { exchange } from "./refresh.js";
+import { throwFormattedError } from "./utils.js";
+
+/**
+ * Send a security code to a phone number
+ * @param {String} firstFactorCode
+ * @param {String} to
+ * @returns {Object}
+ */
+export async function sendSecurityCodeSms({ firstFactorCode, to } = {}) {
+  if (!firstFactorCode || !to) {
+    throw new Error("Userfront.sendSecurityCodeSms missing parameters.");
+  }
+
+  return sendSecurityCode({
+    firstFactorCode,
+    to,
+    strategy: "securityCode",
+    channel: "sms",
+  });
+}
+
+/**
+ * Send an MFA security code
+ * @param {String} firstFactorCode
+ * @param {String} strategy
+ * @param {String} channel
+ * @param {String} to
+ * @returns {Object}
+ */
+export async function sendSecurityCode({
+  firstFactorCode,
+  strategy,
+  channel,
+  to,
+} = {}) {
+  if (!firstFactorCode || !strategy || !channel || !to) {
+    throw new Error("Userfront.sendSecurityCode missing parameters.");
+  }
+
+  try {
+    const { data } = await axios.post(`${store.baseUrl}auth/mfa`, {
+      tenantId: store.tenantId,
+      firstFactorCode,
+      strategy,
+      channel,
+      to,
+    });
+
+    return data;
+  } catch (error) {
+    throwFormattedError(error);
+  }
+}
+
+/**
+ * Submit a phone number and MFA security code
+ * @param {String} to
+ * @param {String} securityCode
+ * @param {String|Boolean} redirect Redirect to given path unless specified as `false`
+ * @returns {Object}
+ */
+export async function loginWithSecurityCode({
+  to,
+  securityCode,
+  redirect,
+} = {}) {
+  if (!to || !securityCode) {
+    throw new Error("Userfront.loginWithSecurityCode missing parameters.");
+  }
+
+  try {
+    const { data } = await axios.put(`${store.baseUrl}auth/mfa`, {
+      tenantId: store.tenantId,
+      to,
+      securityCode,
+    });
```

Comment:
This does not follow the [spec](https://www.notion.so/userfront/MFA-9a69437c382e416fb20671f8a652159e#25a24ca97d26429aaeffd385e505ab75) as designed, which I'm assuming means the endpoint also doesn't follow the spec.

Why is `to` still needed here, and why is the `firstFactorCode` not included?

Reply:
My mistake. I should've just rewritten the tests. This isn't even the correct payload.